aspnetcore/test/hot-reload.md

```diff
@@ -0,0 +1,68 @@
+---
+title: .NET Hot Reload support for ASP.NET Core
+author: rick-anderson
+description: Use .NET Hot Reload to apply code changes to a running app without restarting the app and without losing app state.
+monikerRange: '>= aspnetcore-6.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 10/27/2021
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: test/hot-reload
+---
+# .NET Hot Reload support for ASP.NET Core
+
+.NET Hot Reload applies code changes, including changes to stylesheets, to a running app without restarting the app and without losing app state. Hot Reload is supported for all ASP.NET Core projects.
+
+Generally, updated code is rerun to take effect with the following conditions:
+
+* Some startup logic is only run once:
+  * Middleware, unless the code update is to an inline middleware delegate.
+  * Configured services.
+  * Route creation and configuration, unless the code update is to a route handler delegate.
```

Comment:
Should we also include `OnInitialized` as an example here since we've had lots of people hit that particular scenario.

Reply:
```suggestion
* Route creation and configuration, unless the code update is to a route handler delegate (for example, `OnInitialized`).
```